lib/file.js

```diff
@@ -165,12 +162,16 @@ File.prototype.toString = function () {
  * @api private
  */
 
-File.prototype.modified = function (path) {
+File.prototype.modified = function *() {
   try {
-    return stat(path).mtime.getTime();
+    var stat = yield fs.stat(this.path);
+    this.attrs.mtime = stat.mtime.getTime();
   } catch (e) {
-    return (new Date()).getTime();
+    var now = new Date();
+    this.attrs.mtime = now.getTime();
```

Comment:
I just thought it was more consistent with the block above, no other reason than that.